core/src/smartcontracts/isi/triggers.rs

```diff
@@ -174,18 +172,16 @@ pub mod isi {
                             false
                         };
                     if allow_execute {
-                        acc.lock().unwrap().push((id.clone(), authority.clone()));
                         Ok(())
                     } else {
                         Err(ValidationError::new("Unauthorized trigger execution").into())
                     }
                 })
-                .ok_or_else(|| Error::Find(Box::new(FindError::Trigger(id))))?;
-            let new_array = acc.lock().unwrap().clone();
-            for (id, authority) in new_array {
```

Comment:
This function had kinda strange implementaion previously, it seems like `execute_trigger` could be called more then once, however `acc` is empty or contain only one item.

Reply:
Yes I should change it to an `Option`. This design should change long term though.

Reply:
> This function had kinda strange implementaion previously, it seems like `execute_trigger` could be called more then once, however `acc` is empty or contain only one item.

It's what I did when rebasing. I didn't know if the closure got called more then once. It's one of the many reasons I don't like closures.

Reply:
> closure got called more then once
@SamHSmith, i think `rust` do us good service here, it provides `FnOnce` closure type which might be called only once, however in this paticular case  inside `inspect_by_id` `Fn` is used which is misleading.